### PR TITLE
Add Kafka BOM to Micronaut 3.5.0

### LINKED ISSUE #7116: Add kafka BOM to core
### Issue description

https://github.com/micronaut-projects/micronaut-kafka/issues/509

### PR DESCRIPTION
**Pull request by timyates**
[skip ci]

Blocked until https://github.com/micronaut-projects/micronaut-kafka/pull/510 is released as 4.3.0

### CODE REVIEW

File: gradle/libs.versions.toml
Comment (sdelamo) [inline gradle/libs.versions.toml:259]: why do we have this? Do we know?

**Review by sdelamo — APPROVED** [review]

File: gradle/libs.versions.toml
Comment (timyates) [reply gradle/libs.versions.toml:259]: I have no idea...  I originally deleted it, but I guess someone may be consuming it as a dependency, so I added it back in...

@graemerocher Do you recall where this is required?  I can find it added in 2018 here https://github.com/micronaut-projects/micronaut-core/commit/0f8c3f6682eb7d285af3acb7f8fd727ee116b7a7#diff-49a96e7eea8a94af862798a45174e6ac43eb4f8b4bd40759b5da63ba31ec3ef7R80

My vote would be to delete it for 3.5.0, and see who squeaks

Comment (melix) [reply gradle/libs.versions.toml:259]: I don't think that `kafka212` is a great name either. Obviously encoding the version in the name of the alias kind of defeats the purpose. Maybe this should be reworked for 4.0.

**Review by sdelamo — APPROVED** [review]
I have released [Kafka 4.3.0](https://github.com/micronaut-projects/micronaut-kafka/releases/tag/v4.3.0). It should be good to go once CI passes. .

**Review by melix — APPROVED** [review]

**Comment by timyates** [comment]
@sdelamo I can't merge it, as the CLA isn't triggering 😢

**Pull request merged**
